test/SlowTests/Issues/RavenDB-21566.cs

```diff
@@ -0,0 +1,232 @@
+            }
+        }
+    }
+
+    private class Dto
+    {
+        public string Id { get; set; }
+        public bool? NullableBool { get; set; }
+        public int? NullableInt { get; set; }
+        public decimal? NullableDecimal { get; set; }
+        public double? NullableDouble { get; set; }
+        public float? NullableFloat { get; set; }
+    }
+
+    private class DtoWithNullableEnum
+    {
+        public DummyEnum? NullableEnum { get; set; }
+    }
+
+    private class DtoWithNullableDates
+    {
+        public DateTime? NullableDateTime { get; set; }
+        public TimeSpan? NullableTimeSpan { get; set; }
+        public DateTimeOffset? NullableDateTimeOffset { get; set; }
+        public DateOnly? NullableDateOnly { get; set; }
+        public TimeOnly? NullableTimeOnly { get; set; }
+    }
+    
+    private enum DummyEnum
+    {
+        Value1,
+        Value2
+    }
+
+    private class DummyIndex : AbstractIndexCreationTask<Dto>
+    {
+        public class IndexResult
+        {
+            public string Id { get; set; }
+            public bool NonNullableBool { get; set; }
+            public int NonNullableInt { get; set; }
+            public decimal NonNullableDecimal { get; set; }
+            public double NonNullableDouble { get; set; }
+            public float NonNullableFloat { get; set; }
+        }
+        
+        public DummyIndex()
+        {
+            Map = dtos => from dto in dtos
+                select new IndexResult()
+                {
+                    Id = dto.Id, 
+                    NonNullableBool = dto.NullableBool.GetValueOrDefault(),
+                    NonNullableInt = dto.NullableInt.GetValueOrDefault(),
+                    NonNullableDecimal = dto.NullableDecimal.GetValueOrDefault(),
+                    NonNullableDouble = dto.NullableDouble.GetValueOrDefault(),
+                    NonNullableFloat = dto.NullableFloat.GetValueOrDefault()
+                };
+            
+            StoreAllFields(FieldStorage.Yes);
+        }
+    }
+    
+    private class DummyIndexWithNullableEnum : AbstractIndexCreationTask<DtoWithNullableEnum>
+    {
+        public class IndexResult
+        {
+            public DummyEnum NonNullableEnum { get; set; }
+        }
+        
+        public DummyIndexWithNullableEnum()
+        {
+            Map = dtos => from dto in dtos
+                select new IndexResult()
+                {
+                    NonNullableEnum = dto.NullableEnum.GetValueOrDefault()
+                };
+            
+            StoreAllFields(FieldStorage.Yes);
+            
+            AdditionalSources = new Dictionary<string, string>
+            {
+                {
+                    "DummyAdditionalSource", 
+                    @"
+                    using System;
+
+                    namespace Temp 
+                    {
+                        public enum DummyEnum 
```

Comment:
Why do we define it?

Reply:
Now it's used to check if this change works for type defined through additional sources and registered on server using conventions